### DISCUSSION TITLE
xterm5.3.0 replaces the first character of a line when the input character is too long 当输入字符过长时候会他替换行首字符

### DISCUSSION THREAD

**Discussion by yanceysong** [sourcelair/xterm.js#4892] [Q&A] [answered]
我在使用Xterm5.3.0库的时候发现，当我终端输入的长度大于某一个值的时候，新输入的字符并不会随着我设置的cols起效，而是会替换行首的字符。麻烦问一下各位如何解决？
When I was using the Xterm5.3.0 library, I found that when the length of my terminal input was greater than a certain value, the newly entered character did not take effect with the cols I set, but replaced the character at the beginning of the line. Could you tell me how to solve it?

下面是我的演示gif
Here is my demo gif
![2](https://github.com/xtermjs/xterm.js/assets/60731354/9f12c7ab-e2d9-4e8c-b1c9-df20694cc936)

下面是我的实例化代码
Here is my instantiation code
```javascript
const term = new Terminal({
            cols: 240,
            rows: 100,
            cursorBlink: true,
            convertEol: false, 
            cursorStyle: "block",
            scrollback: 1, 
            tabStopWidth: 8, 
            wraparoundMode: true,
            theme: {
                foreground: '#5eb66b',
                background: '#030c26', 
                cursor: 'help',
            },
            rightClickSelectsWord: true,
            fontSize: 18,
            screenKeys: true
        });

**Comment by tisilent** [comment]
Cannot repro. 😐

**Comment by yanceysong** [comment, marked as answer]
这个问题我解决了，是因为jsch开启Linux终端的时候有限制，导致前端在超过80字符的时候，Linux会给一个换行和空字符串。解决办法是在开启终端以后执行stty cols 240，或者channel进行设置        
 ((ChannelShell) channel).setPtyType("xterm");
 ((ChannelShell) channel).setPtySize(5000, 5000, 640, 480);  // 设置列数为 5000



I solved this problem because jsch had a restriction when opening the Linux terminal, which caused the front-end to give a newline and an empty string when it exceeded 80 characters. The workaround is to run stty cols 240 after opening the terminal, or set the channel ((ChannelShell) channel).setPtyType("xterm");
((ChannelShell) channel).setPtySize(5000, 5000, 640, 480); // Set the number of columns to 5000

**Reply by yanceysong** [reply]
似乎写死5000不是一个好的选择，这个数值应该是前端xterm获取到然后重新定义

It seems that writing down 5000 is not a good choice, this value should be obtained by the front-end xterm and then redefined

**Comment by MrLeungE1** [comment]
您好，我想问您一个问题，您在开发中是否遇到过，当输入命令过长(超过一行)，对其进行删除的时候，光标的位置停留到了第二行的起始位置不会改变的问题？如果有的话，能帮忙解惑吗？

**Reply by yanceysong** [reply]
似乎和我的出现的问题不一致，你这个我不是很清楚不好意思

**Reply by MrLeungE1** [reply]
感谢你的恢复，我已经通过ANS编码解决了这个问题。

**Reply by carolin-violet** [reply]
请问咋解决的呀
